syntax = "proto3";
package metric;




import "collector_service_sdk/model/collector_service/aggregate_metric_rule.proto";

/*
ListCollectorAggregateMetricRule请求
*/
message ListCollectorAggregateMetricRuleRequest {
    /*
    页码
    */
    int32 page = 1 ;
    /*
    每页大小
    */
    int32 page_size = 2 ;
    /*
    聚合规则名称
    */
    string name = 3 ;
    /*
    聚合指标所属模型id
    */
    string metricOwner = 4 ;
    /*
    聚合指标数据来源模型id
    */
    string metricSource = 5 ;
}

/*
ListCollectorAggregateMetricRule返回
*/
message ListCollectorAggregateMetricRuleResponse {
    /*
    页码
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated collector_service.CollectorAggregateMetricRule list = 4 ;
}

/*
ListCollectorAggregateMetricRuleApi返回
*/
message ListCollectorAggregateMetricRuleResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    metric.ListCollectorAggregateMetricRuleResponse data = 4 ;
}

